pkg/fmap/fmap.go

```diff
@@ -127,36 +143,57 @@ func Read(f io.Reader) (*FMap, *Metadata, error) {
 		return nil, nil, err
 	}
 
-	// Check for too many fmaps.
-	if bytes.Count(data, Signature) >= 2 {
-		return nil, nil, errors.New("Found multiple signatures")
-	}
+	// Loop over __FMAP__ occurrences until a valid header is found
+	start := 0
+	ValidFmaps := 0
+	var fmap FMap
+	var fmapMetadata Metadata
+	for {
+		if start >= len(data) {
+			break
+		}
 
-	// Check for too few fmaps.
-	start := bytes.Index(data, Signature)
-	if start == -1 {
-		return nil, nil, errors.New("Cannot find fmap signature")
-	}
+		next := bytes.Index(data[start:], Signature)
+		if next == -1 {
+			break
+		}
+		start += next
+
+		if start >= len(data) {
```

Comment:
Is this ever false?

Reply:
bytes.Index would have to be wrong in some way?